data/ha/hawk_test.py

```diff
@@ -0,0 +1,829 @@
+        RETURN_VALUE = 23
+
+def ssh_verify_node_maintenance(ssh):
+    global RETURN_VALUE
+    if check_cluster_conf_ssh(ssh, "crm status | grep -i ^node", "maintenance"):
+        print("INFO: cluster node set successfully in maintenance mode")
+        set_test_status('set_first_node_maintenance', 'passed')
+    else:
+        print("ERROR: cluster node failed to switch to maintenance mode")
+        set_test_status('set_first_node_maintenance', 'failed')
+        RETURN_VALUE = 24
+
+def ssh_verify_primitive(ssh, myprimitive, version):
+    global RETURN_VALUE
+    matches = ["%s anything" % str(myprimitive), "binfile=file", "op start timeout=35s",
+               "op monitor timeout=9s interval=13s", "meta target-role=Started"]
+    if Version(checkstr(version)) < Version('15'):
+        matches.append("op stop timeout=15s")
+    else:
+        matches.append("op stop timeout=15s on-fail=stop")
+    if check_cluster_conf_ssh(ssh, "crm configure show", matches):
+        print("INFO: primitive [%s] correctly defined in the cluster configuration" % myprimitive)
+        set_test_status('add_primitive', 'passed')
+    else:
+        print("ERROR: primitive [%s] missing from cluster configuration" % myprimitive)
+        set_test_status('add_primitive', 'failed')
+        RETURN_VALUE = 25
+
+def ssh_verify_primitive_removed(ssh):
+    global RETURN_VALUE
+    if check_cluster_conf_ssh(ssh, "crm resource list | grep ocf::heartbeat:anything", ''):
+        print("INFO: primitive successfully removed")
+        set_test_status('remove_primitive', 'passed')
+    else:
+        print("ERROR: primitive [%s] still present in the cluster while checking with SSH" %
+              myprimitive)
+        set_test_status('remove_primitive', 'failed')
+        RETURN_VALUE = 26
+
+### MAIN
+
+# Command line argument parsing
+parser = argparse.ArgumentParser(description='HAWK GUI interface Selenium test')
+parser.add_argument('-b', '--browser', type=str,
+                    help='Browser to use in the test. Can be: firefox, chrome, chromium')
+parser.add_argument('-H', '--host', type=str, default='localhost',
+                    help='Host or IP address where HAWK is running')
+parser.add_argument('-P', '--port', type=str, default='7630',
+                    help='TCP port where HAWK is running')
+parser.add_argument('-p', '--prefix', type=str, default='',
+                    help='Prefix to add to Resources created during the test')
+parser.add_argument('-t', '--test-version', type=str, default='',
+                    help='Test version. Ex: 12-SP3, 12-SP4, 15, 15-SP1')
+parser.add_argument('-s', '--secret', type=str, default='',
+                    help='root SSH Password of the HAWK node')
+parser.add_argument('-r', '--results', type=str, default='',
+                    help='Generate hawk_test.results file')
+args = parser.parse_args()
+if not args.browser:
+    print('ERROR: must specify a browser with --browser')
```

Comment:
ArgumentParser has special property of argument something like required= true . this will allow you to drop this if at all and let ArgumentParser do the job

Reply:
Fixing it.